README.md

```diff
@@ -115,6 +115,57 @@ Encoding the payload can be done with the [Ruby `jwt` gem](https://github.com/jw
 JWT.encode payload, SECRET, 'HS256'
 ```
 
+## Settings vs config vs Environment variables
+
+```bash
+config/settings.yml
+# load order 1,  basic default config settings
+config/settings/#{environment}.yml
+# load order 2, environment config settings
+config/environments/#{environment}.yml
+# load order 3, environment config settings based on one for the environments (either development|test|producation)
+printenv | grep SETTINGS
+# load order 4, Machine based env variables settings
+```
+
+All basic default settings are to be listed in `config/settings.yml`,
+which in order can be overridden by `config/settings/#{environment}.yml`,
+and in turn overrided by `config/environments/#{environment}.yml`,
+and finally overrided by `Machine based env variables settings`
+
+To override it via `Machine based env variables settings`
+```bash
+cat config/settings.yml
+file
+  based
+    setting
+      env1: 'some file based value'
+```
+
+```bash
+env SETTINGS__file__based__setting__env1=`machine wins"
+```
+
+```ruby
+print Settings.file.based.setting.env1
+
+# machine wins
+```
+
+Any `Machine based env variables settings` that is not prefixed with `SETTINGS`.* are not considered for general consumption.
+
+## Linting
```

Comment:
No need to reference the linting here, it's already in the readme

Reply:
Argh bad merge